scripts/src/admin/wrapper.html

```diff
@@ -0,0 +1,365 @@
+		}
+		</style>
+		<link href="https://cdn.kinmarketplace.com/admin/jsoneditor.min.css" rel="stylesheet" type="text/css">
+	</head>
+	<body>
+
+		<h1><a href="/">Marketplace Admin</a></h1>
+		<div id="toast">MSG TOAST</div>
+		<div class="overlay"">
+			<div class="overlay-content flex-row-container">
+				<div class="text" class="wide"></div>
+				<div class="controls">
+					<button class="preview-btn btn">Refresh Preview</button>
+					<br />
+					<button class="publish-btn btn">Publish</button>
+					<br />
+					<button class="btn" onclick="overlayOff()">Close</button>
+				</div>
+				<div class="preview"><iframe></iframe></div>
+			</div>
+		</div>
+		<div id="content">${ content }</div>
+		<div id="footer"></div>
+
+		<script src="https://cdnjs.cloudflare.com/ajax/libs/axios/0.18.0/axios.min.js" async></script>
+		<script src="https://cdn.kinmarketplace.com/admin/jsoneditor.min.js" async></script>
+
+		<script>
+			let overlayElement;
+			let overlayTextElement;
+			let toastElemnt;
+			let previewElemnt;
+			let previewBtnElemnt;
+			let publishBtnElemnt;
+			var overlayJsonEditor;
+
+			document.addEventListener("DOMContentLoaded", function() {
+				overlayElement = document.querySelector(".overlay");
+				overlayTextElement = document.querySelector(".overlay .text");
+				previewBtnElemnt = document.querySelector(".overlay .preview-btn");
+				previewElemnt = document.querySelector(".overlay .preview iframe");
+				publishBtnElemnt = document.querySelector(".overlay .publish-btn");
+
+				toastElemnt = document.querySelector("#toast");
+				previewBtnElemnt.addEventListener("click", refreshPreviewBtnHandler)
+				publishBtnElemnt.addEventListener("click", publishBtnHandler)
+			});
+
+			function toast(msg) {
+				toastElemnt.innerText = msg;
+				toastElemnt.className = "show";
+				setTimeout(function(){ toastElemnt.className = toastElemnt.className.replace("show", ""); }, 3000);
+			}
+			function submitData(url, data) {
+				axios.post(url, data)
+					.then(res => toast("ok"))
+					.catch(err => alert("error: " + JSON.stringify(err)));
+			}
+			function overlayOn(text, offerId) {
+				var jsonEditorOptions = {
+					mode: 'form',
+					modes: ['view', 'form', 'tree', 'code']
+
+				};
+				overlayElement.style.display = "block";
+				overlayElement.dataset.offerId = offerId;
+
+				overlayJsonEditor = new JSONEditor(overlayTextElement, jsonEditorOptions, JSON.parse(unescape(text), null,  2));
+				overlayJsonEditor.expandAll();
+			}
+			function overlayOff() {
+				overlayElement.style.display = "none";
+				overlayTextElement.textContent = "";
+				previewElemnt.src = "";
+			}
+
+			function stringifyJson(json){
+				// This isn't really needed because the JSON editor escapes and protects but just to be on the safe side
+				try {
+					jsonData = JSON.stringify(json);
+					return jsonData;
+				} catch {
+					toast("Invaid JSON");
+					return false;
+				}
+			}
+
+			function editAppConfigs(json, appId) {

```

Comment:
Whats this? Is this needed?

Reply:
@Doc999tor do we need both this function and the below `openEditor(jsonArea, appId)` function?